src/gatling/scala/SingleLeaseMultipleChecks.scala

```diff
@@ -0,0 +1,45 @@
+import com.typesafe.config.{Config, ConfigFactory}
+import com.warrenstrange.googleauth.GoogleAuthenticator
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+
+import scala.concurrent.duration._
+
+class SingleLeaseMultipleChecks extends Simulation {
+  val config: Config = ConfigFactory.load()
+
+  val authenticator: GoogleAuthenticator = new GoogleAuthenticator()
+
+  val requestJwt = exec(
+    http("Lease token")
+      .post("/lease")
+      .formParam("microservice", config.getString("service.name"))
+      .formParam("oneTimePassword", "${otp}")
+      .check(bodyString.saveAs("jwt"))
+  )
+
+  val checkJwt = exec(
+    http("Check token")
+      .get("/details")
+      .header("Authorization", "Bearer ${jwt}")
+      .check(substring(config.getString("service.name")))
+  )
+
+  val otpFeeder = Iterator.continually(Map("otp" -> authenticator.getTotpPassword(config.getString("service.pass"))))
+
+  setUp(
+    scenario("Testing")
+      .feed(otpFeeder)
+      .exec(requestJwt)
+      .repeat(config.getInt("repeats")) {
```

Comment:
It may be better to replace the repeats with a duration-based solution. We know that the token will become invalid within 30 seconds, but we don't know how many checks that will allow us to do.

Also, we're ramping users over 1 minute, but the scenarios can't take longer than 30 seconds (the expiry of the key), meaning we'll never see the maximum number of users hitting s2s simultaneously. In order to make the duration of a scenario less constrained, we need to request a new token every now and then (perhaps by introducing a top-level repeat?).